### ISSUE TITLE
viscious.widgets.bat (dis)charge rate is always N/A

### ISSUE THREAD

**Issue by arch-stack** [vicious-widgets/vicious#119]
When I use viscious.widgets.bat on linux, $5 the battery (dis)charge rate is always N/A

I am using a framework 13 laptop with arch linux and it appears whichever battery module is loaded is not providing the power_now file that is used in https://github.com/vicious-widgets/vicious/blob/master/widgets/bat_linux.lua#L49C5-L51C8

While

> /sys/class/power_supply/BAT1/power_now

is missing

> /sys/class/power_supply/BAT1/current_now
> /sys/class/power_supply/BAT1/voltage_now

are present, which means we should be able to use those as a fallback

We could change:

```
if battery.power_now then
    curpower = string.format("%.2f", tonumber(battery.power_now) /1000000)
end
```

To:

```
if battery.power_now then
    curpower = string.format("%.2f", tonumber(battery.power_now) /1000000)
elseif battery.current_now and battery.voltage_now then
    curpower = string.format("%.2f", tonumber(battery.current_now) * tonumber(battery.voltage_now) / 10^12)
end
```

**Comment by McSinyx** [comment]
Thank you for reporting.  Your patch seems reasonable, could you please [open a PR](https://vicious.readthedocs.io/en/latest/contributing.html#requesting-for-merging-a-patch)?

**Comment by arch-stack** [comment]
I just about have a PR ready, the only thing is I only have it tested in lua 5.3 as that appears to be what arch has awesomewm compiled against

Should be fine for such a small change based on existing code?

**Comment by arch-stack** [comment]
I have opened a PR for this issue

So far no issues during my occasional usage over the last week

**Comment by McSinyx** [comment]
Thanks, I've just released v2.7.1 including your patch in case you want to update Vicious for your distro.
